### PR TITLE
cleanup kubectl apiresources

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:
#42873 introduce this new subcommand.

This PR does
* binding flags to struct
* some code cleanups

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
/cc soltysh juanvallejo 

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by dixudx** [comment]
cc @kubernetes/sig-cli-api-reviews PTAL

**Comment by dixudx** [comment]
@liggitt @deads2k PTAL. Thanks.

**Comment by dixudx** [comment]
/assign @liggitt @deads2k 
PTAL. Thanks.

File: pkg/kubectl/cmd/apiresources.go [hunk 1]
```diff
@@ -181,7 +170,16 @@ func (o *ApiResourcesOptions) RunApiResources(cmd *cobra.Command, f cmdutil.Fact
 		}
 	}
 
-	sort.Stable(sortableGroupResource(resources))
+	sort.Slice(resources, func(i, j int) bool {
```
Comment (deads2k) [inline pkg/kubectl/cmd/apiresources.go]: I prefer this as a separate named type.

File: pkg/kubectl/cmd/apiresources.go [hunk 2]
```diff
@@ -85,37 +86,25 @@ func NewCmdApiResources(f cmdutil.Factory, ioStreams genericclioptions.IOStreams
...
 
-	cmd.Flags().Bool("no-headers", false, "When using the default or custom-column output format, don't print headers (default print headers).")
-	cmd.Flags().StringP("output", "o", "", "Output format. One of: wide|name.")
+	cmd.Flags().BoolVar(&o.NoHeaders, "no-headers", o.NoHeaders, "When using the default or custom-column output format, don't print headers (default print headers).")
+	cmd.Flags().StringVarP(&o.Output, "output", "o", o.Output, "Output format. One of: wide|name.")
 
-	cmd.Flags().StringVar(&o.APIGroup, "api-group", "", "Limit to resources in the specified API group.")
-	cmd.Flags().BoolVar(&o.Namespaced, "namespaced", true, "Namespaced indicates if a resource is namespaced or not.")
+	cmd.Flags().StringVar(&o.APIGroup, "api-group", o.APIGroup, "Limit to resources in the specified API group.")
+	cmd.Flags().BoolVar(&o.Namespaced, "namespaced", o.Namespaced, "Namespaced indicates if a resource is namespaced or not.")
```
Comment (juanvallejo) [inline pkg/kubectl/cmd/apiresources.go]: Maybe this description could be improved a bit? Something like `Namespaced filters resources by the given namespace` or something similar

**Comment by dixudx** [comment]
@deads2k @juanvallejo Updated. PTAL. Thanks.

**Comment by dixudx** [comment]
@deads2k Your suggestion sound good to me. Will do it later.

**Pull request merged**